packages/json/src/rules/keyDuplicates.ts

```diff
@@ -25,18 +25,20 @@ export default jsonLanguage.createRule({
 			],
 		},
 	},
-	options: {
-		allowKeys: z
-			.array(z.string())
-			.default([])
-			.describe("Keys to allow duplicates under."),
-	},
-	setup(context, { allowKeys }) {
+	options: z
+		.object({
+			allowKeys: z
+				.array(z.string())
+				.default([])
+				.describe("Keys to allow duplicates under."),
+		})
+		.optional(),
```

Comment:
`options` is now a Zod schema itself instead of a `Record` of Zod schemas because that was a) easier and b) seemed more flexible. However, it does mean that you need a Zod dependency for rules with no options (`z.never()`), which isn't an issue in practice but is a tad annoying and for the idiomatic partial object configuration, you'll need to set optional yourself, which might be a footgun in practice.